### PR TITLE
Use renderer more in circuit construction manual

### PR DESCRIPTION
**Pull request by CalMacCQ**
# Description

I've done a pass over the `Circuit construction` section of the manual using the circuit renderer in place of `circ.get_commands()`

In fact a subsection on the renderer should probably appear early in the manual #331. 

I've also made some changes to the section on `CircBox` particularly with the `Circuit.add_circbox_with_regmap` and `Circuit.add_circbox_regwise` methods.

I think some of the code examples need to be updated too. Maybe thats best done in another PR.

### CODE REVIEW

File: manual/manual_circuit.rst [hunk 1]
```diff
@@ -586,6 +586,8 @@ Boxes
 
 Working with individual basic gates is sufficient for implementing arbitrary circuits, but that doesn't mean it is the most convenient option. It is generally far easier to argue the correctness of a circuit's design when it is constructed using higher-level constructions. In ``pytket``, the concept of a "Box" is to abstract away such complex structures as black-boxes within larger circuits.
 
+Defining higher level subroutines as boxes is also beneficial from a circuit optimisation point of view. If the compiler can identify higher level structure in the circuit, this can be exploited to reduce the number of elementary gates in the compiled circuit. Examples of such optimisations can be seen in :py:class:`~pytket.circuit.ToffoliBox` which permutes the computational basis states and :py:class:`~pytket.circuit.ConjugationBox` which allows for more efficient controlled gates by exploiting circuit symmetry.
```
Comment (CalMacCQ) [inline manual/manual_circuit.rst:589]: Thought I should add this extra paragraph as boxes in pytket are not just useful from the point of view of arguing circuit correctness. In several cases we have a higher level construct which can be implemented efficently by the compiler.

File: manual/manual_circuit.rst [hunk 2]
```diff
@@ -1087,7 +1118,9 @@ Looking at the statevector calculation we see that our :py:class:`~pytket.circui
 Importing/Exporting Circuits
 ----------------------------
 
-A ``pytket`` :py:class:`~pytket.circuit.Circuit` s can be natively serializaed and deserialized from JSON-compatible dictionaries, using the :py:meth:`~pytket.circuit.Circuit.to_dict` and :py:meth:`~pytket.circuit.Circuit.from_dict` methods. This is the method of serialization which supports the largest class of circuits, and provides the highest fidelity.
+.. currentmodule:: pytket.circuit
+
+A ``pytket`` :py:class:`~pytket.circuit.Circuit` s can be natively serializaed and deserialized from JSON-compatible dictionaries, using the :py:meth:`Circuit.to_dict` and :py:meth:`Circuit.from_dict` methods. This is the method of serialization which supports the largest class of circuits, and provides the highest fidelity.
```
Comment (willsimmons1465) [inline manual/manual_circuit.rst:1123]:
```suggestion
``pytket`` :py:class:`~pytket.circuit.Circuit` s can be natively serialized and deserialized from JSON-compatible dictionaries, using the :py:meth:`Circuit.to_dict` and :py:meth:`Circuit.from_dict` methods. This is the method of serialization which supports the largest class of circuits, and provides the highest fidelity.
```

File: manual/manual_circuit.rst [hunk 3]
```diff
@@ -863,13 +889,17 @@ To understand what happens inside a :py:class:`~pytket.circuit.PauliExpBox` let'
...
+These Pauli gadget circuits have interesting algebraic properties which are useful for circuit optimisation. 
+
+A Pauli gadget can be expressed as :math:`V \, A \, V^\dagger` where :math:`V` is the the circuit composed of CX gates and single qubit basis rotations on the right hand side of the Rz gate and :math:`A` is the Rz gate itself. This observation allows one to construct controlled Pauli gadgets much more efficently. See the `blog post <https://tket.quantinuum.com/blog/posts/controlled_gates/>`_ on the :py:class:`~pytket.circuit.ConjugationBox` construct for more details.
+
+For further discussion see the research publication on phase gadget synthesis [Cowt2020]_. Ideas from this paper are implemented in TKET as the `OptimisePhaseGadgets <https://tket.quantinuum.com/api-docs/passes.html#pytket.passes.OptimisePhaseGadgets>` and `PauliSimp <https://tket.quantinuum.com/api-docs/passes.html#pytket.passes.PauliSimp>`_ optimisation passes.
```
Comment (willsimmons1465) [inline manual/manual_circuit.rst:902]:
```suggestion
For further discussion see the research publication on phase gadget synthesis [Cowt2020]_. Ideas from this paper are implemented in TKET as the `OptimisePhaseGadgets <https://tket.quantinuum.com/api-docs/passes.html#pytket.passes.OptimisePhaseGadgets>`_ and `PauliSimp <https://tket.quantinuum.com/api-docs/passes.html#pytket.passes.PauliSimp>`_ optimisation passes.
```

File: manual/manual_circuit.rst [hunk 4]
```diff
@@ -1652,14 +1703,16 @@ To add gates or boxes to a circuit with specified op group names, simply pass th
...
     newcirc = Circuit(1).Rx(0.125, 0).Ry(0.875, 0)
     circ.substitute_named(newcirc, "rotations")
+
     # Replace the "special one" with a different op:
     newop = Op.create(OpType.T)
     circ.substitute_named(newop, "special one")
+
     # Substitute a box for the "Fred" group:
-    newcbox = CircBox(Circuit(2).H(1).CX(1,0))
+    newcbox = CircBox(Circuit(2, name="C'").H(1).CX(1, 0))
```
Comment (willsimmons1465) [inline manual/manual_circuit.rst:1712]: C' is probably not as clear to read in this context as a different letter

**Review by willsimmons1465 — COMMENTED** [review]
One formatting error, one typo, and one minor comment. Otherwise this is a very good change that helps readability and approachability.

In reading through the build, I also spotted an additional formatting error on line 1317 (mixing URL and directive links for KAKDecomposition)

File: manual/manual_circuit.rst (hunk 4 above)
Comment (CalMacCQ) [reply manual/manual_circuit.rst]: I've opted for `P` and `Q` instead. Hopefully this is clearer for the reader.

**Comment by CalMacCQ** [comment]
> One formatting error, one typo, and one minor comment. Otherwise this is a very good change that helps readability and approachability.
> 
> In reading through the build, I also spotted an additional formatting error on line 1317 (mixing URL and directive links for KAKDecomposition)

Thanks for the review :) I've made the suggested changes and fixed the KAK referencing mistake.

**Review by willsimmons1465 — APPROVED** [review]
Looks good now, thanks for the edits!

**Pull request merged**